source/Ship.cpp

```diff
@@ -583,6 +585,11 @@ void Ship::FinishLoading(bool isNewInstance)
 			thumbnail = model->thumbnail;
 	}
 
+	for(auto &hardpoint : armament.GetEditable())
+	{
+		hardpoint.AddOffset(center);
+	}
```

Comment:
Instead of doing this make Armament responsible for setting the offsets for the hardpoints. So like here it would call `armament.SetOffset(center);` for example.

Reply:
I moved it to armament.FinishLoading, I hope that's ok